invenio_rdm_records/services/access/service.py

```diff
@@ -130,6 +130,13 @@ def _validate_secret_link_expires_at(
         was set in the given data, or if it was omitted (which makes a
         difference in patch operations).
         """
+
+        if current_app.config.get("RDM_RECORDS_REQUIRE_SECRET_LINKS_EXPIRATION") and not expires_at:
```

Comment:
I would suggest avoiding the optional look up key `.get()` when we expect to be defined (True or False), so that we can fail fast if there is a misconfiguration, rather than a feature silently disabled:
```suggestion
        if current_app.config["RDM_RECORDS_REQUIRE_SECRET_LINKS_EXPIRATION"] and not expires_at:
```